### PR TITLE
fstring in io.sql

### PR DESCRIPTION
**Pull request by lexy-lixinyu**
- [ ] ref #29547
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Review by WillAyd — CHANGES_REQUESTED** [review]
Thanks for the PR. Looks good and I think CI failures are unrelated / have been fixed on master since this was created.

To be sure, can you locally on your `shiny-new-feature` branch run:

```sh
git fetch upstream
git merge upstream/master
git push origin shiny-new-feature
```

I think should get CI green for this. Flag me down at PyData if you have any questions

File: pandas/io/sql.py [hunk 1]
```diff
@@ -825,8 +821,8 @@ def _index_name(self, index, index_label):
                     index_label = [index_label]
                 if len(index_label) != nlevels:
                     raise ValueError(
-                        "Length of 'index_label' should match number of "
-                        "levels, which is {0}".format(nlevels)
+                        f"Length of 'index_label' should match number of "
+                        "levels, which is {nlevels}"
```
Comment (simonjayhawkins) [inline pandas/io/sql.py]: this line needs an `f`

File: pandas/io/sql.py [hunk 2]
```diff
@@ -1304,10 +1300,7 @@ def to_sql(
 
             for col, my_type in dtype.items():
                 if not isinstance(to_instance(my_type), TypeEngine):
-                    raise ValueError(
-                        "The type of {column} is not a "
-                        "SQLAlchemy type ".format(column=col)
-                    )
+                    raise ValueError(f"The type of {col} is not a " "SQLAlchemy type ")
```
Comment (simonjayhawkins) [inline pandas/io/sql.py]:
```suggestion
                    raise ValueError(f"The type of {col} is not a SQLAlchemy type ")
```

File: pandas/io/sql.py [hunk 3]
```diff
@@ -1599,14 +1588,11 @@ def execute(self, *args, **kwargs):
                 self.con.rollback()
             except Exception as inner_exc:  # pragma: no cover
                 ex = DatabaseError(
-                    "Execution failed on sql: {sql}\n{exc}\nunable "
-                    "to rollback".format(sql=args[0], exc=exc)
+                    f"Execution failed on sql: {args[0]}\n{exc}\nunable " "to rollback"
```
Comment (simonjayhawkins) [inline pandas/io/sql.py]:
```suggestion
                    f"Execution failed on sql: {args[0]}\n{exc}\nunable to rollback"
```

File: pandas/io/sql.py [hunk 4]
```diff
@@ -1731,11 +1717,7 @@ def to_sql(
         if dtype is not None:
             for col, my_type in dtype.items():
                 if not isinstance(my_type, str):
-                    raise ValueError(
-                        "{column} ({type!s}) not a string".format(
-                            column=col, type=my_type
-                        )
-                    )
+                    raise ValueError(f"{col} ({my_type!s}) not a string")
```
Comment (simonjayhawkins) [inline pandas/io/sql.py]: Is the `!s` required here?

**Comment by WillAyd** [comment]
@lexy-lixinyu can you commit changes as suggested by @simonjayhawkins ?

File: pandas/io/sql.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/io/sql.py]: remove trailing space?

**Comment by jbrockmendel** [comment]
just pushed edits removing unintentional " " in two places.

**Comment by datapythonista** [comment]
Thanks for the work on this @lexy-lixinyu, can you address the two last comments, and update your branch from master please? This is ready to be merged after it.

If you need help let us know.

**Comment by jbrockmendel** [comment]
just pushed edits to address outstanding comments

**Review by datapythonista — APPROVED** [review]
lgtm

**Comment by simonjayhawkins** [comment]
Thanks @lexy-lixinyu and @jbrockmendel

**Pull request merged**